File: backend/components/metiers/administration_publique/services/core/samples/sample_service.test.py
"""
sample_service.test.py
Tests unitaires pour SampleService (clé en main, CI/CD ready)
"""
import pytest
from .sample_service import SampleService

def test_sample_service_init():
    service = SampleService(options={"mode": "test"})
    assert service.options == {"mode": "test"}
    assert service.state == 'ready'

def test_sample_service_config():
    service = SampleService()
    assert service.init({"level": 1}) is True
    assert service.config == {"level": 1}
    assert service.state == 'initialized'

def test_sample_service_run():
    service = SampleService()
    service.init({"level": 2})
    result = service.run("DATA")
    assert result == {"processed": True, "data": "DATA", "config": {"level": 2}, "state": "initialized"}

File: backend/components/metiers/administration_publique/services/core/samples/sample_service.test.js
// sample_service.test.js
// Tests unitaires pour SampleService (clé en main, CI/CD ready)

const SampleService = require('./sample_service');

describe('SampleService', () => {
  it('doit s\'initialiser correctement', () => {
    const service = new SampleService({ mode: 'test' });
    expect(service.options).toEqual({ mode: 'test' });
    expect(service.state).toBe('ready');
  });

  it('doit initialiser la config', () => {
    const service = new SampleService();
    expect(service.init({ level: 1 })).toBe(true);
    expect(service.config).toEqual({ level: 1 });
    expect(service.state).toBe('initialized');
  });

  it('doit exécuter le run sample', () => {
    const service = new SampleService();
    service.init({ level: 2 });
    const result = service.run('DATA');
    expect(result).toEqual({ processed: true, data: 'DATA', config: { level: 2 }, state: 'initialized' });
  });
});
